src/compas_model/elements/element.py

```diff
@@ -5,723 +5,402 @@
+    surface_area : float
+        The area of the geometry. Measurement is made from the ``geometry`` meshes.
     volume : float
-        the volume of an element based on geometry 3d geometry
+        The volume of the geometry. Measurement is made from the ``geometry`` meshes.
+    center_of_mass : :class:`compas.geometry.Point`
+        The center of mass of the geometry. Measurement is made from the ``geometry`` meshes.
+    centroid : :class:`compas.geometry.Point`
+        The centroid of the geometry. Measurement is made from the ``geometry`` meshes.
+    is_support : bool
+        Indicates whether the element is a support.
+    forces : list[tuple]
+        A list of Tuple(name, lines) that can be used for force vector vizualization.
+    fabrication : list[tuple]
+        A list of Tuple(name, geometry) information that can be used for fabrication geometry vizualization.
 
     Examples
     --------
-    >>> element = Element("BLOCK", 0, Frame.worldXY(), Point(0,0,0), \
-                        Box( \
-                            frame=Frame([3, 0, 0], [0.866, 0.1, 0.0], [0.5, 0.866, 0.0]), \
-                            xsize=2, ysize=4, zsize=0.25), \
-                        [Vector(0, 0, 1), 0])
+    >>> frame = Frame([3, 0, 0], [0.866, 0.1, 0.0], [0.5, 0.866, 0.0])
+    >>> box = Box(frame=frame, xsize=2, ysize=4, zsize=0.25)
+    >>> element = Element("BLOCK", Frame.worldXY(), Point(0,0,0), box)
+    >>> element.insertion = Vector(0, 0, 1)
 
     """
 
     def __init__(
         self,
         name=None,
-        id=None,
         frame=None,
         geometry_simplified=None,
         geometry=None,
-        insertion=None,
+        copy_geometry=True,
         **kwargs
     ):
         # --------------------------------------------------------------------------
-        # call the inherited Data constructor for json serialization
+        # Call the inherited Data constructor.
         # --------------------------------------------------------------------------
         super(Element, self).__init__()
 
         # --------------------------------------------------------------------------
-        # name
-        # the string can be any string
-        # but better use the existing container: compas_assembly2.ELEMENT_NAME.BLOCK
-        # --------------------------------------------------------------------------
-        if not name:
-            self.name = "CUSTOM"
-        else:
-            self.name = name.upper()
-
-        # --------------------------------------------------------------------------
-        # indexing
-        # indices are store as a list to keep grouping information e.g. [0,1]
+        # Set the element name.
         # --------------------------------------------------------------------------
-        if id is None:
-            self.id = [-1]
-        else:
-            self.id = [id] if isinstance(id, int) else id
+        self.name = name.lower() if name else ElementType.UNKNOWN
```

Comment:
i guess this is why you gave your enumeration string values but that is not a good reason :)

Reply:
i don't see an added value compared to the naming mechanism already inherited from data

Reply:
Removed, made it simple, without any Enum, there was no use case.